internal/execute/tableutil/stream.go

```diff
@@ -0,0 +1,184 @@
+package tableutil
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+)
+
+// SendFunc is a function that will be used to construct a stream.
+type SendFunc func(flux.ColReader)
+
+// Stream will call StreamContext with a background context.
+func Stream(f func(ctx context.Context, fn SendFunc) error) (flux.Table, error) {
+	return StreamContext(context.Background(), f)
+}
+
+// StreamContext will create a table that streams column readers
+// through the flux.Table. This method will return only after
+// the function buffers the first column reader.
+// This first column reader is used to identify the group key
+// and columns for the entire table stream.
+//
+// Implementors using this *must* return at least one table.
+// If the function returns without returning at least one table,
+// then an error will be returned. If the first table that is returned
+// is empty, then this will return an empty table and further buffers
+// will not be used.
+func StreamContext(ctx context.Context, f func(ctx context.Context, fn SendFunc) error) (flux.Table, error) {
```

Comment:
It sounds more like `StreamWithContext`. At first glance, I thought this is to create some context called `StreamContext`

Reply:
I'll change the name. I thought I was mimicking parts of the standard library, but I see that it uses `WithContext` for what I thought.